### PR TITLE
Update hex-fiend to 2.8.0

### PR DESCRIPTION
**Pull request by reitermarkus**
After making all changes to the cask:

- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] The commit message includes the cask’s name and version.

### CODE REVIEW

File: Casks/hex-fiend.rb [hunk 1]
```diff
@@ -5,7 +5,7 @@
   # github.com/ridiculousfish/HexFiend was verified as official when first introduced to the cask
   url "https://github.com/ridiculousfish/HexFiend/releases/download/v#{version}/Hex_Fiend_#{version.major_minor}.dmg"
   appcast 'https://github.com/ridiculousfish/HexFiend/releases.atom',
-          checkpoint: 'a08db7b7bd0ea6df35d193897ac5c5b11ed894703d9df5f67dfbd2d3828060de'
+          checkpoint: 'cd523fdd9bb2b5222dfd5b276aa1cfdb10dfa3ed0e798d78ba3fc771d7208478'
```
Comment (commitay) [inline Casks/hex-fiend.rb]: Why?

Comment (reitermarkus) [reply Casks/hex-fiend.rb]: Why what?

Comment (commitay) [reply Casks/hex-fiend.rb]: Why send a PR for a checkpoint only change?

To be clear: I'm not objecting to maintainers doing it as they can self merge but we don't want contributors to see PRs for checkpoint only changes and start submitting them again.

Comment (reitermarkus) [reply Casks/hex-fiend.rb]: > Why send a PR for a checkpoint only change?

I was checking my installed Casks for outdated appcasts, so might as well update them.

> we don't want contributors to see PRs for checkpoint only changes and start submitting them again.

When was this discussed? Seems I have forgotten.

Comment (commitay) [reply Casks/hex-fiend.rb]: https://github.com/Homebrew/homebrew-cask/issues/38376, https://github.com/Homebrew/homebrew-cask/issues/43605

Comment (vitorgalvao) [reply Casks/hex-fiend.rb]: Might be a good time to mention I’ve started working on new scripts a while back (there’s an issue somewhere with @colindunn), though I haven’t yet had the time to expand on them. But I did prefer the method I was employing, in which I’d somehow (depends on the cask) use the appcasts to figure out a version, format it in the same way we used it on the cask, then compare it to the cask’s version.

I’m thinking about removing `checkpoint`s completely in favour of that method.

Comment (commitay) [reply Casks/hex-fiend.rb]: > (there’s an issue somewhere with @colindunn)

https://github.com/Homebrew/homebrew-cask/issues/41240

> I’m thinking about removing `checkpoint`s completely in favour of that method.

👍 

I've never used checkpoints for their intended purpose of finding outdated casks, I only find them useful sometimes as a way of seeing if the appcast has changed when an PR is submitted e.g. `bartender` and people submitting dev updates.

Comment (commitay) [reply Casks/hex-fiend.rb]: Also, removing checkpoints would mean we can use unstable appcasts and we don't have to deal with CI checkpoint failures.

**Pull request merged**